data/attributes.go

```diff
@@ -0,0 +1,42 @@
+package data
+
+import "strings"
+
+const (
+	attributesSeparator = ";"
+	keyValuesSeparator  = ":"
+	valuesSeparator     = ","
+)
+
+// Attributes hold data about token attributes
+type Attributes map[string][]string
+
+// NewAttributesDTO will create a new instance of Attributes base of the input attributes
+func NewAttributesDTO(attributes []byte) *Attributes {
+	if len(attributes) == 0 {
+		return nil
+	}
+
+	attrs := make(Attributes)
+	sAttributes := strings.Split(string(attributes), attributesSeparator)
+
+	if len(sAttributes) == 1 {
+		attrs[string(attributes)] = []string{string(attributes)}
```

Comment:
why returning directly here?
Does this work properly if the input string is
`shard:1,3` ? I suggest to remove the if branch from L23
Are we assuming that all provided attributes buffer values end up with `;`?

Reply:
refactored. changed the logic. good catch.